FreeRTOS-Plus/Demo/Common/Logging/logging_stack.h

```diff
@@ -40,8 +40,17 @@
 #include <stdint.h>
 
 /* Metadata information to prepend to every log message. */
+
+/* Macro to extract only the file name from file path to use for metadata in
+ * log messages. */
+#ifdef _MSC_VER
+    #define FILENAME           ( strrchr( __FILE__, '\\' ) ? ( strrchr( __FILE__, '\\' ) + 1 ) : __FILE__ )
```

Comment:
I do not like this. Firstly it severely increases the cost of logging, instead of passing a constant pointer around you now do string manipulation every time. Also ironically strrchr is not supposed to be used as per MISRA Rule 21.17 which is a mandatory rule.